// react/lib/ReactIsomorphic.js
define('node_modules/react/lib/ReactIsomorphic', function(require, exports, module) {

  var process = require('node_modules/process/browser');
  /**
   * Copyright 2013-2015, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   * @providesModule ReactIsomorphic
   */
  
  'use strict';
  
  var ReactChildren = require('node_modules/react/lib/ReactChildren');
  var ReactComponent = require('node_modules/react/lib/ReactComponent');
  var ReactClass = require('node_modules/react/lib/ReactClass');
  var ReactDOMFactories = require('node_modules/react/lib/ReactDOMFactories');
  var ReactElement = require('node_modules/react/lib/ReactElement');
  var ReactElementValidator = require('node_modules/react/lib/ReactElementValidator');
  var ReactPropTypes = require('node_modules/react/lib/ReactPropTypes');
  var ReactVersion = require('node_modules/react/lib/ReactVersion');
  
  var assign = require('node_modules/react/lib/Object.assign');
  var onlyChild = require('node_modules/react/lib/onlyChild');
  
  var createElement = ReactElement.createElement;
  var createFactory = ReactElement.createFactory;
  var cloneElement = ReactElement.cloneElement;
  
  if (process.env.NODE_ENV !== 'production') {
    createElement = ReactElementValidator.createElement;
    createFactory = ReactElementValidator.createFactory;
    cloneElement = ReactElementValidator.cloneElement;
  }
  
  var React = {
  
    // Modern
  
    Children: {
      map: ReactChildren.map,
      forEach: ReactChildren.forEach,
      count: ReactChildren.count,
      toArray: ReactChildren.toArray,
      only: onlyChild
    },
  
    Component: ReactComponent,
  
    createElement: createElement,
    cloneElement: cloneElement,
    isValidElement: ReactElement.isValidElement,
  
    // Classic
  
    PropTypes: ReactPropTypes,
    createClass: ReactClass.createClass,
    createFactory: createFactory,
    createMixin: function (mixin) {
      // Currently a noop. Will be used to validate and trace mixins.
      return mixin;
    },
  
    // This looks DOM specific but these are actually isomorphic helpers
    // since they are just generating DOM strings.
    DOM: ReactDOMFactories,
  
    version: ReactVersion,
  
    // Hook for JSX spread, don't use this for anything else.
    __spread: assign
  };
  
  module.exports = React;

});
