// react/lib/ReactInjection.js
define('node_modules/react/lib/ReactInjection', function(require, exports, module) {

  
  /**
   * Copyright 2013-2015, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   * @providesModule ReactInjection
   */
  
  'use strict';
  
  var DOMProperty = require('node_modules/react/lib/DOMProperty');
  var EventPluginHub = require('node_modules/react/lib/EventPluginHub');
  var ReactComponentEnvironment = require('node_modules/react/lib/ReactComponentEnvironment');
  var ReactClass = require('node_modules/react/lib/ReactClass');
  var ReactEmptyComponent = require('node_modules/react/lib/ReactEmptyComponent');
  var ReactBrowserEventEmitter = require('node_modules/react/lib/ReactBrowserEventEmitter');
  var ReactNativeComponent = require('node_modules/react/lib/ReactNativeComponent');
  var ReactPerf = require('node_modules/react/lib/ReactPerf');
  var ReactRootIndex = require('node_modules/react/lib/ReactRootIndex');
  var ReactUpdates = require('node_modules/react/lib/ReactUpdates');
  
  var ReactInjection = {
    Component: ReactComponentEnvironment.injection,
    Class: ReactClass.injection,
    DOMProperty: DOMProperty.injection,
    EmptyComponent: ReactEmptyComponent.injection,
    EventPluginHub: EventPluginHub.injection,
    EventEmitter: ReactBrowserEventEmitter.injection,
    NativeComponent: ReactNativeComponent.injection,
    Perf: ReactPerf.injection,
    RootIndex: ReactRootIndex.injection,
    Updates: ReactUpdates.injection
  };
  
  module.exports = ReactInjection;

});
