// react/lib/ReactDOM.js
define('node_modules/react/lib/ReactDOM', function(require, exports, module) {

  var process = require('node_modules/process/browser');
  var global = typeof global !== "undefined" ? global : typeof self !== "undefined" ? self : typeof window !== "undefined" ? window : {};
  /**
   * Copyright 2013-2015, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   * @providesModule ReactDOM
   */
  
  /* globals __REACT_DEVTOOLS_GLOBAL_HOOK__*/
  
  'use strict';
  
  var ReactCurrentOwner = require('node_modules/react/lib/ReactCurrentOwner');
  var ReactDOMTextComponent = require('node_modules/react/lib/ReactDOMTextComponent');
  var ReactDefaultInjection = require('node_modules/react/lib/ReactDefaultInjection');
  var ReactInstanceHandles = require('node_modules/react/lib/ReactInstanceHandles');
  var ReactMount = require('node_modules/react/lib/ReactMount');
  var ReactPerf = require('node_modules/react/lib/ReactPerf');
  var ReactReconciler = require('node_modules/react/lib/ReactReconciler');
  var ReactUpdates = require('node_modules/react/lib/ReactUpdates');
  var ReactVersion = require('node_modules/react/lib/ReactVersion');
  
  var findDOMNode = require('node_modules/react/lib/findDOMNode');
  var renderSubtreeIntoContainer = require('node_modules/react/lib/renderSubtreeIntoContainer');
  var warning = require('node_modules/react/node_modules/fbjs/lib/warning');
  
  ReactDefaultInjection.inject();
  
  var render = ReactPerf.measure('React', 'render', ReactMount.render);
  
  var React = {
    findDOMNode: findDOMNode,
    render: render,
    unmountComponentAtNode: ReactMount.unmountComponentAtNode,
    version: ReactVersion,
  
    /* eslint-disable camelcase */
    unstable_batchedUpdates: ReactUpdates.batchedUpdates,
    unstable_renderSubtreeIntoContainer: renderSubtreeIntoContainer
  };
  
  // Inject the runtime into a devtools global hook regardless of browser.
  // Allows for debugging when the hook is injected on the page.
  /* eslint-enable camelcase */
  if (typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ !== 'undefined' && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.inject === 'function') {
    __REACT_DEVTOOLS_GLOBAL_HOOK__.inject({
      CurrentOwner: ReactCurrentOwner,
      InstanceHandles: ReactInstanceHandles,
      Mount: ReactMount,
      Reconciler: ReactReconciler,
      TextComponent: ReactDOMTextComponent
    });
  }
  
  if (process.env.NODE_ENV !== 'production') {
    var ExecutionEnvironment = require('node_modules/react/node_modules/fbjs/lib/ExecutionEnvironment');
    if (ExecutionEnvironment.canUseDOM && window.top === window.self) {
  
      // First check if devtools is not installed
      if (typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ === 'undefined') {
        // If we're in Chrome or Firefox, provide a download link if not installed.
        if (navigator.userAgent.indexOf('Chrome') > -1 && navigator.userAgent.indexOf('Edge') === -1 || navigator.userAgent.indexOf('Firefox') > -1) {
          console.debug('Download the React DevTools for a better development experience: ' + 'https://fb.me/react-devtools');
        }
      }
  
      // If we're in IE8, check to see if we are in compatibility mode and provide
      // information on preventing compatibility mode
      var ieCompatibilityMode = document.documentMode && document.documentMode < 8;
  
      process.env.NODE_ENV !== 'production' ? warning(!ieCompatibilityMode, 'Internet Explorer is running in compatibility mode; please add the ' + 'following tag to your HTML to prevent this from happening: ' + '<meta http-equiv="X-UA-Compatible" content="IE=edge" />') : undefined;
  
      var expectedFeatures = [
      // shims
      Array.isArray, Array.prototype.every, Array.prototype.forEach, Array.prototype.indexOf, Array.prototype.map, Date.now, Function.prototype.bind, Object.keys, String.prototype.split, String.prototype.trim,
  
      // shams
      Object.create, Object.freeze];
  
      for (var i = 0; i < expectedFeatures.length; i++) {
        if (!expectedFeatures[i]) {
          console.error('One or more ES5 shim/shams expected by React are not available: ' + 'https://fb.me/react-warning-polyfills');
          break;
        }
      }
    }
  }
  
  module.exports = React;

});
