// react/lib/ReactDefaultInjection.js
define('node_modules/react/lib/ReactDefaultInjection', function(require, exports, module) {

  var process = require('node_modules/process/browser');
  /**
   * Copyright 2013-2015, Facebook, Inc.
   * All rights reserved.
   *
   * This source code is licensed under the BSD-style license found in the
   * LICENSE file in the root directory of this source tree. An additional grant
   * of patent rights can be found in the PATENTS file in the same directory.
   *
   * @providesModule ReactDefaultInjection
   */
  
  'use strict';
  
  var BeforeInputEventPlugin = require('node_modules/react/lib/BeforeInputEventPlugin');
  var ChangeEventPlugin = require('node_modules/react/lib/ChangeEventPlugin');
  var ClientReactRootIndex = require('node_modules/react/lib/ClientReactRootIndex');
  var DefaultEventPluginOrder = require('node_modules/react/lib/DefaultEventPluginOrder');
  var EnterLeaveEventPlugin = require('node_modules/react/lib/EnterLeaveEventPlugin');
  var ExecutionEnvironment = require('node_modules/react/node_modules/fbjs/lib/ExecutionEnvironment');
  var HTMLDOMPropertyConfig = require('node_modules/react/lib/HTMLDOMPropertyConfig');
  var ReactBrowserComponentMixin = require('node_modules/react/lib/ReactBrowserComponentMixin');
  var ReactComponentBrowserEnvironment = require('node_modules/react/lib/ReactComponentBrowserEnvironment');
  var ReactDefaultBatchingStrategy = require('node_modules/react/lib/ReactDefaultBatchingStrategy');
  var ReactDOMComponent = require('node_modules/react/lib/ReactDOMComponent');
  var ReactDOMTextComponent = require('node_modules/react/lib/ReactDOMTextComponent');
  var ReactEventListener = require('node_modules/react/lib/ReactEventListener');
  var ReactInjection = require('node_modules/react/lib/ReactInjection');
  var ReactInstanceHandles = require('node_modules/react/lib/ReactInstanceHandles');
  var ReactMount = require('node_modules/react/lib/ReactMount');
  var ReactReconcileTransaction = require('node_modules/react/lib/ReactReconcileTransaction');
  var SelectEventPlugin = require('node_modules/react/lib/SelectEventPlugin');
  var ServerReactRootIndex = require('node_modules/react/lib/ServerReactRootIndex');
  var SimpleEventPlugin = require('node_modules/react/lib/SimpleEventPlugin');
  var SVGDOMPropertyConfig = require('node_modules/react/lib/SVGDOMPropertyConfig');
  
  var alreadyInjected = false;
  
  function inject() {
    if (alreadyInjected) {
      // TODO: This is currently true because these injections are shared between
      // the client and the server package. They should be built independently
      // and not share any injection state. Then this problem will be solved.
      return;
    }
    alreadyInjected = true;
  
    ReactInjection.EventEmitter.injectReactEventListener(ReactEventListener);
  
    /**
     * Inject modules for resolving DOM hierarchy and plugin ordering.
     */
    ReactInjection.EventPluginHub.injectEventPluginOrder(DefaultEventPluginOrder);
    ReactInjection.EventPluginHub.injectInstanceHandle(ReactInstanceHandles);
    ReactInjection.EventPluginHub.injectMount(ReactMount);
  
    /**
     * Some important event plugins included by default (without having to require
     * them).
     */
    ReactInjection.EventPluginHub.injectEventPluginsByName({
      SimpleEventPlugin: SimpleEventPlugin,
      EnterLeaveEventPlugin: EnterLeaveEventPlugin,
      ChangeEventPlugin: ChangeEventPlugin,
      SelectEventPlugin: SelectEventPlugin,
      BeforeInputEventPlugin: BeforeInputEventPlugin
    });
  
    ReactInjection.NativeComponent.injectGenericComponentClass(ReactDOMComponent);
  
    ReactInjection.NativeComponent.injectTextComponentClass(ReactDOMTextComponent);
  
    ReactInjection.Class.injectMixin(ReactBrowserComponentMixin);
  
    ReactInjection.DOMProperty.injectDOMPropertyConfig(HTMLDOMPropertyConfig);
    ReactInjection.DOMProperty.injectDOMPropertyConfig(SVGDOMPropertyConfig);
  
    ReactInjection.EmptyComponent.injectEmptyComponent('noscript');
  
    ReactInjection.Updates.injectReconcileTransaction(ReactReconcileTransaction);
    ReactInjection.Updates.injectBatchingStrategy(ReactDefaultBatchingStrategy);
  
    ReactInjection.RootIndex.injectCreateReactRootIndex(ExecutionEnvironment.canUseDOM ? ClientReactRootIndex.createReactRootIndex : ServerReactRootIndex.createReactRootIndex);
  
    ReactInjection.Component.injectEnvironment(ReactComponentBrowserEnvironment);
  
    if (process.env.NODE_ENV !== 'production') {
      var url = ExecutionEnvironment.canUseDOM && window.location.href || '';
      if (/[?&]react_perf\b/.test(url)) {
        var ReactDefaultPerf = require('node_modules/react/lib/ReactDefaultPerf');
        ReactDefaultPerf.start();
      }
    }
  }
  
  module.exports = {
    inject: inject
  };

});
